forest_height.R

```diff
@@ -0,0 +1,25 @@
+# Function to decide height of forest plots based on the number of studies
+# and the output format
+# 
+# param studies
+# A number (an int). The number of studies to be plotted.
+# param tab 
+# A string representing the way the plot height is measured
+# "pixel" for png files or plots rendered in the app 
+# "inch" for pdf files
+#
+# return 
+# A number 
+# Height in pixels for png plots
+# Height in inches for pdf plots
+#
+# author
+# NVB
+
+forest_height <- function(studies, measure){
+  if (measure == "pixel") {
+    ifelse(studies <=20, 400, 20*studies)
+  } else if (measure == "inch") {
+    ifelse(studies <=20, 6, 6 + 0.2*(studies-20))
```

Comment:
Is there a reason why the 2 measurement systems don't scale the same way?
When the inch height is calculated for more than 20 studies, it increases 1/30th of the baseline height per additional study, whereas the pixel height increases by 1/20th of the baseline height.

Reply:
Calculations edited to make the scaling more in proportion (although not exact)

For pixels it is now: studies <=20, 400, 400 + 15*(studies-20)

At more than 20 studies the rendered images within the app are squashed. A fixed minimum size of 400 pixels allows for the forest plots to be displayed within the app without needing to scroll. The smaller increment of 15 pixels per study still allows for forest plots with >20 studies to be displayed without the studies being too close.

For pdfs it is now: studies <=25, 6, 6 + 0.2*(studies-25)

This is taken from the MetaInsight code to produce pdf plots and produces forest plots correctly spaced in the 20-30 study range (and above)